pkg/controller/storagecluster/reconcile.go

```diff
@@ -275,6 +290,55 @@ func (r *ReconcileStorageCluster) Reconcile(request reconcile.Request) (reconcil
 	return reconcile.Result{}, phaseErr
 }
 
+// getNodeTopologyMap
+func (r *ReconcileStorageCluster) getNodeTopologyMap(sc *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	nodes := &corev1.NodeList{}
+	nodeListOptions := client.ListOptions{}
+	nodeListOptions.SetLabelSelector(nodeAffinityKey)
+
+	err := r.client.List(context.TODO(), &nodeListOptions, nodes)
+	if err != nil {
+		return err
+	}
+
+	topologyMap := ocsv1.NodeTopologyMap{}
+
+	for _, node := range nodes.Items {
+		labels := node.Labels
+		for _, label := range labels {
+			for _, key := range validTopologyLabelKeys {
+				if strings.Contains(label, key) {
```

Comment:
Because `validTopologyLabelKeys` is something like `failure-domain.kubernetes.io` and not the full label, e.g. `failure-domain.kubernetes.io/zone`.

Reply:
I think we should put these consts in validTopologyLabelKeys: https://github.com/kubernetes/api/blob/master/core/v1/well_known_labels.go#L21

If  we want to support arbitrary topology keys, we should have restrictive regex for them. Allowing this might have unintended consequences from label collisions. Infra labels should either come as documented behavior from openshift/kubernetes, or be prefixed with the ocs domain.

Reply:
Maybe it would be clearer to say I don't want to hijack the `failure-domain.kubernetes.io/xxx` with ocs-specific behaviour. Someone else might have the same idea and cause a conflict. If we are planning to allow that, we should do `failure-domain..ocs.openshift.io/xxx`

Reply:
Yeah, that's fine. We're just working with what we got for now.